### PR TITLE
Extend consensus alignment to local analysis

### PR DESCRIPTION
**Pull request by fede-pe**
This PR includes:

- Modification of consensus alignment to allow local analysis.

### CODE REVIEW

File: tomo/protocols/protocol_ts_consensus_alignment.py [hunk 1]
```diff
@@ -308,6 +332,166 @@ def compareTransformationMatrices(Mset, shiftTol, angleTol, SRset):
...
             return averageAlignmentV, angleSDV, shiftSDV
 
+    @staticmethod
+    def compareTransformationMatricesLocal(Mset, shiftTol, angleTol, SRset):
+        print("Running local consensus alignment...")
+
+        Nts = len(Mset)
+        Nti = len(Mset[0])
+
+        print("Number of tilt-series analyzed: " + str(Nts))
```
Comment (pconesa) [inline tomo/protocols/protocol_ts_consensus_alignment.py:342]: We are removing all prints alternatives are:

self.log("blah, blah, ...") being self a protocol instance

or


import logging
logger logging.getLogger(__name__)

...
...
logger.info("blah, blah, ...")
logger.debug("blah, blah, ...")
logger.warining("blah, blah, ...")
logger.error("blah, blah, ...")

File: tomo/protocols/protocol_ts_consensus_alignment.py [hunk 2]
```diff
@@ -308,6 +332,166 @@ def compareTransformationMatrices(Mset, shiftTol, angleTol, SRset):
...
+                tiConsensusAlignment /= numberOfConsensusAlignment
+                averageAlignmentV.append(tiConsensusAlignment)
+
+        # Check if all images are misaligned
+        for a in averageAlignmentV:
+            if a is not None:
+                return averageAlignmentV, angleSDV, shiftSDV
+
+        return None, None, None
+
```
Comment (pconesa) [inline tomo/protocols/protocol_ts_consensus_alignment.py:494]: in the generateTsIdLIst is much faster, although here there will be not much gain to use

set.getIdSet() at least for the Tiltimages -->https://github.com/scipion-em/scipion-pyworkflow/blob/devel/pyworkflow/object.py#L1307

And to get the list of tsId you could use an aggregation function:

https://github.com/scipion-em/scipion-pyworkflow/blob/devel/pyworkflow/object.py#L1358

getUniqueValues()

Comment (pconesa) [reply tomo/protocols/protocol_ts_consensus_alignment.py:494]: set(self.inputMultiSoTS[0].getUniqueValues('_tsId'))

**Comment by pconesa** [comment]
Thanks Fede. Only thing is that logger is better to be declare at module level (not protocol class) and use logger = logging.get_logger(__name_)

**Comment by fede-pe** [comment]
I think I fixed it as expected.

**Review by pconesa — APPROVED** [review]

**Pull request merged**